src/main/java/touro/spf/SPFGraphView.java

```diff
@@ -1,4 +1,52 @@
 package touro.spf;
 
-public class SPFGraphView {
+import javax.swing.*;
+import java.awt.*;
+import java.util.HashSet;
+
+public class SPFGraphView extends JComponent {
+
+    Graph graph;
+    private int WIDTH_HEIGHT = 50;
+
+    public SPFGraphView(Graph graph) {
+        this.graph = graph;
+    }
+
+    @Override
+    protected void paintComponent(Graphics g) {
+        super.paintComponent(g);
+        paintGraph(g);
+    }
+
+    private void paintGraph(Graphics g) {
+        HashSet<Node> nodes = new HashSet<>();
```

Comment:
Yes I am realizing this will be helpful to have more than once. Is the method buildGraph() similar to what we want if we just change it to return the nodes instead of it being void?